lib/oban/telemetry.ex

```diff
@@ -69,6 +69,82 @@ defmodule Oban.Telemetry do
   * `:conf` — the config of the Oban supervised producer
   * `:dispatched_count` — the number of jobs fetched and started by the producer
 
+  ### Engine Events
+
+  Oban emits telemetry span events for the following Engine operations:
```

Comment:
The `init`, `refresh`, and `put_meta` events may also write to the database. So you'll need to wrap those as well to cover all of the interactions.